diff -ru okular-22.04.1.a/CMakeLists.txt okular-22.04.1.b/CMakeLists.txt
--- okular-22.04.1.a/CMakeLists.txt	2022-05-10 07:47:22.000000000 +0200
+++ okular-22.04.1.b/CMakeLists.txt	2022-08-08 01:04:44.512748733 +0200
@@ -69,7 +69,7 @@
                   VERSION_HEADER "${CMAKE_CURRENT_BINARY_DIR}/core/version.h"
                   PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/Okular5ConfigVersion.cmake")
 
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED COMPONENTS Core Test Widgets PrintSupport Svg Qml Quick)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED COMPONENTS Core Test Widgets PrintSupport Svg Qml Quick Test)
 if (BUILD_DESKTOP)
     find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED COMPONENTS DBus)
 endif()
